### DISCUSSION TITLE
Django project openai integration 504 timeout.

### DISCUSSION THREAD

**Discussion by hengjiUSTC** [vercel/community#2069] [Help] [answered]
### Problem Description

For https://github.com/vercel/examples/blob/main/python/django/vercel.json we have use case that some operations (open ai apis) takes longer than 10s. Currently vercel returns 504 error similar to https://community.openai.com/t/api-504s-in-production-vercel-only/28795/8. I tried to update vercel.json to

```

{
  "builds": [
    {
      "src": "vercel_app/wsgi.py",
      "use": "@vercel/python"
    }
  ],
  "routes": [
    {
      "src": "/(.*)",
      "dest": "vercel_app/wsgi.py"
    }
  ],
  "functions": {
    "bot/**/*": {
      "maxDuration": 60
    }
  }
}
```
But this won't work due to conflict of functions and builds. How could we fix this issue to enlarge timeout limit ?

**Comment by amyegan** [comment, marked as answer]
Hi @hengjiUSTC! The [serverless execution timeout limit](https://vercel.com/docs/concepts/limits/overview) is determined by the account type. It is 10 seconds for Hobby accounts, 60 seconds for Pro teams, and 900 seconds for Enterprise teams.

**Reply by sarangzambare** [reply]
Facing similar issue. Just started the Pro trial and transferred my projects. Still getting 504. Does it take a while for the timeout change to take place?

**Reply by Divokg** [reply]
Same issue here, I have the Pro package. The time limit defaults to 15 seconds. but when you increase maxDuration to 60 on the vercel.json file. It doesnt execute. Still timesout at 15 seconds

**Comment by ms8909** [comment]
Cant increase maxDuration in Django.

Create a simple python function in Vercel and add the following in vercel.json file

"functions": {
"api/*.py": {
"maxDuration": 300
}

**Reply by NiccoloKadera** [reply]
I tried doing so in a Django instance, but received error "deployment failed". I can confirm I have /api folder, and I'm using vercel hobby plan which maximum limit should be 60s.

"functions": {
   "api/*.py": {
      "maxDuration": 15
   }
 }
